packages/muelu/src/Graph/Containers/MueLu_Aggregates_kokkos_def.hpp

```diff
@@ -260,6 +260,43 @@ namespace MueLu {
     return vertex2AggId_->getMap();
   }
 
+  template <class LocalOrdinal, class GlobalOrdinal, class DeviceType>
+  void
+  Aggregates_kokkos<LocalOrdinal, GlobalOrdinal, Kokkos::Compat::KokkosDeviceWrapperNode<DeviceType> >::PrintAllNodesPerAggregate(
+    Teuchos::FancyOStream& out, bool useGlobalNodeIds) const
+  {
+    // Compute list of nodes per aggregate
+    LO_view aggPtr;
+    LO_view aggNodes;
+    LO_view unaggregated;
+    ComputeNodesInAggregate(aggPtr, aggNodes, unaggregated);
+
+    // Loop aggregates and their nodes and print them
+    const int myRank = vertex2AggId_->getMap()->getComm()->getRank();
+    LO numAggregates = GetNumAggregates();
+    const auto aggSizes = ComputeAggregateSizes(true);
+    TEUCHOS_ASSERT(numAggregates==aggSizes.size());
+
+    std::stringstream myStream;
+    myStream << "\n";
+    for (LO lAggId = Teuchos::OrdinalTraits<LO>::zero(); lAggId < numAggregates; ++lAggId)
```

Comment:
@cgcgcg In the Kokkos version, I should probably do a `Kokkos::parallel_for` here, right? However, then the output might be scrambled, so how can I obtain deterministic output for the unit test?

Reply:
Since this is a print routine, I'd `deep_copy` to host and do the printing on host since you can't print from device anyway

Reply:
Ok, I have added my approach to this. Not sure, if its the best approach, though.